### PR TITLE
test: clean up git perf asserts and migrate to new harness

### PR DESCRIPTION
**Pull request by kaihowl**
## Summary

- Refactors and standardizes git-perf Bash tests to the new harness and assertion helpers.
- Cleans up brittle stdout parsing by using test_section blocks and assertion-based validation.
- Adds and relies on common helpers (e.g., assert_success_with_output, assert_contains, assert_not_contains, assert_equals, assert_matches, assert_file_exists, etc.).
- Introduces test_stats at the end of each test file to summarize results.
- Normalizes test setup/teardown for consistency.
- Fix: Adjust test-coverage reporting logic in git_perf/src/reporting.rs so HTML and CSV outputs correctly report whether measurements exist. Specifically, finalize() consumption of section outputs is accounted for, and the presence check now uses section content for HTML and the finalized report bytes for CSV.

## Changes

- Migrates 38 Bash tests to the new framework (test_section blocks, centralized test helpers, assertion-driven validation).
- Replaces ad-hoc set -e/set -x usage with the new harness defaults and per-test sections.
- Consolidates stdout validation behind common assertion helpers (e.g., assert_success_with_output, assert_contains, assert_not_contains, assert_equals, assert_matches, assert_file_exists).
- Appends test_stats at the end of each test file to summarize results.
- Normalizes test setup/teardown for consistency across tests.
- Introduces and relies on a uniform testing idiom across affected tests for readability and maintainability.
- Code change: Update in git_perf/src/reporting.rs to correctly detect presence of measurements for HTML vs. CSV outputs by accounting for finalize() consumption and using the appropriate content/bytes checks.
- Test change: Fix missing initialization of cur_ variables in test_remove.sh.

## Rationale

- The new test harness provides reliable, assertion-driven failure messages and reduces brittleness from stdout parsing.
- Centralized helpers simplify maintenance and reduce duplication across tests.
- The reporting-logic fix ensures test outcomes align with actual data presence in HTML/CSV outputs.

## Testing plan

- Run the repository's test suite via CI as usual.
- Validate individual test files under the new harness (e.g., test_config_cmd.sh, test_dispersion_method__.sh, test_report_.sh, test_audit_*.sh).
- Confirm special cases pass: no data scenarios, multiple splits, edge cases for dispersion methods, and correct CSV/HTML reporting behavior.
- Verify that test_stats output is produced and that all assertion helpers report accurate messages.

## Notes for reviewers

- The migration introduces a uniform testing idiom across all affected tests. If CI flags regressions, please investigate environment-specific differences or expectations in the new harness.
- The changes are focused on the testing harness/style and an accompanying fix in the reporting logic; the underlying code under test remains unchanged.
- If preferred, I can split this into smaller PRs (framework migration first, then coverage) for easier review.

🌿 Generated by Terry

📎 **Task**: https://www.terragonlabs.com/task/4d98438d-93b6-47d6-8e25-e1a7ee90f39a

### CODE REVIEW

File: test/test_audit_threshold.sh [hunk 1]
```diff
@@ -19,52 +18,55 @@ min_relative_deviation = 5.0
...
-
-# Test with a measurement that has lower relative deviation
-create_commit
-git perf add -m memory_usage 107
-
-# This should pass due to relative deviation being below 5% global threshold
+assert_success git perf audit -m build_time
+
+test_section "Test global threshold with a different measurement"
+assert_success create_commit
```
Comment (kaihowl) [inline test/test_audit_threshold.sh]: over asserting

File: test/test_size.sh [hunk 2]
```diff
@@ -1,151 +1,135 @@
...
+assert_contains "$output" "Total measurement data size"
 popd
 
-echo "Test 2: Single commit with measurements"
+test_section "Single commit with measurements"
 cd_empty_repo
 create_commit
-git perf add -m test-measure-one 10.0
-git perf add -m test-measure-two 20.0
+assert_success git perf add -m test-measure-one 10.0
```
Comment (kaihowl) [inline test/test_size.sh]: over constrained asserts

File: test/test_operations_without_seed.sh [hunk 3]
```diff
@@ -27,55 +23,61 @@ pushd work_noseed > /dev/null
...
 git push > /dev/null 2>&1
 
-echo "Testing remove operation without seed measurements..."
-# Test remove operation without any measurements (should handle gracefully)
-output=$(git perf remove --older-than '7d' 2>&1 1>/dev/null) && exit_code=0 || exit_code=$?
-if [[ $exit_code -eq 0 ]]; then
-    echo "SUCCESS: Remove operation handled empty measurement set gracefully"
+test_section "Testing remove operation without seed measurements"
+
+# Test remove operation without any measurements (should handle gracefully or fail with expected message)
```
[the remark this answers is not in the document]
Comment (kaihowl) [reply test/test_operations_without_seed.sh]: https://github.com/kaihowl/git-perf/issues/575

File: test/test_remove.sh [hunk 4]
```diff
@@ -116,13 +126,20 @@ if git_objects_contain test-measure-one; then
   exit 1
 fi
 
+git reflog expire --expire=all --all
```
[the remark this answers is not in the document]
Comment (kaihowl) [reply test/test_remove.sh]: correct bug fix
